src/mewbot/io/rss.py

```diff
@@ -0,0 +1,514 @@
+        except AttributeError:
+            problem_list.append("entry did not possess a title attribute")
+            entry_title = TITLE_NOT_SET_STR
+
+        return str(entry_title)
+
+    @staticmethod
+    def _extract_link(entry: feedparser.util.FeedParserDict, problem_list: List[str]) -> str:
+        """
+        Extract and return the link from an entry.
+        Adding any problems to the problem list.
+        """
+        # link
+        try:
+            entry_link = entry.link
+        except AttributeError:
+            problem_list.append("entry did not possess a link attribute")
+            entry_link = LINK_NOT_SET_STR
+
+        return str(entry_link)
+
+    @staticmethod
+    def _extract_description(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        """
+        Extract and return the link from an entry.
+        Adding any problems to the problem list.
+        """
+        # description
+        try:
+            entry_description = entry.description
+        except AttributeError:
+            problem_list.append("entry did not possess a description attribute")
+            entry_description = DESCRIPTION_NOT_SET_STR
+
+        return str(entry_description)
+
+    @staticmethod
+    def _extract_author(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_author = entry.author
+        except AttributeError:
+            problem_list.append("entry did not possess an author attribute")
+            entry_author = AUTHOR_NOT_SET_STR
+
+        return str(entry_author)
+
+    @staticmethod
+    def _extract_category(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_category = entry.category
+        except AttributeError:
+            problem_list.append("entry did not possess a category attribute")
+            entry_category = CATEGORY_NOT_SET_STR
+
+        return str(entry_category)
+
+    @staticmethod
+    def _extract_comments(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_comments = entry.comments
+        except AttributeError:
+            problem_list.append("entry did not possess a comments attribute")
+            entry_comments = COMMENTS_NOT_SET_STR
+
+        return str(entry_comments)
+
+    @staticmethod
+    def _extract_enclosure(entry: feedparser.util.FeedParserDict) -> str:
+        try:
+            entry_enclosure = entry.enclosure
+        except AttributeError:
+            # No logging as this is an optional attribute
+            entry_enclosure = ENCLOSURE_NOT_SET_STR
+
+        return str(entry_enclosure)
+
+    def _extract_guid(
+        self, entry: feedparser.util.FeedParserDict, site_url: str, problem_list: List[str]
+    ) -> str:
+
+        entry_title = self._extract_title(entry=entry, problem_list=problem_list)
```

Comment:
This is only used in the `AttributeError` block, possible doesn't want to be up here